File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, r = 0, n = nums.size(), c = 0, ans = 0;
        while(r<n)
        {
            if(nums[r])
                r++;
            else if(c<k){
                c++;
                r++;
            }
            else
            {
                if(!nums[l])c--;
                l++;
            }
            ans = max(ans, r-l);
        }
        return ans;
    }
    
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l = 0,r = 0,n = nums.length,ans = 0;
        int x = 0;
        while(r<n)
        {
            if(nums[r]==0)x++;
            while(x>k)
            {
                if(nums[l]==0) x--;
                l++;
            }
            ans = Math.max(ans,r-l+1);
            r++;
        }
        return ans;
    }
}